src/main/java/com/algorithmia/client/HttpClient.java

```diff
@@ -172,6 +187,32 @@ private HttpResponse execute(HttpUriRequest request) throws APIException {
         return execute(request, new BasicAsyncResponseConsumer());
     }
 
+    private HttpResponse executeHead(HttpUriRequest request) throws APIException {
+        // We don't use the BasicAsyncResponseConsumer because it barfs when the
```

Comment:
comment doesn't make sense here, this is a minor optimization taking advantage of HEAD requests not returning the body